kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -387,6 +397,14 @@ SetOfKeyValuePairs DBAdapter::lastReachableBlockDbUpdates(const SetOfKeyValuePai
 
   for (const auto &[k, v] : non_provable_kvs) {
     dbUpdates[DBKeyManipulator::genNonProvableBlockDbKey(blockId, k)] = v;
+    if (blockId) {
+      const auto &stale_kv = getValueForNonProvableKey(k, blockId - 1);
+      if (stale_kv) {
+        const auto &nonProvableStaleKey = DBKeyManipulator::genNonProvableStaleDbKey(
+            DBKeyManipulator::genNonProvableBlockDbKey(stale_kv.value().second, k), blockId);
```

Comment:
Nit: I think it will be easier to read if we write these keys on separate lines and naming them. Another thing that comes to mind is that `block` in the name of `genNonProvableBlockDbKey()` might be redundant.

Reply:
Done